### PR TITLE
feat: [CDS-52511]: preparing a skeleton for kubeconfig.

### PR DESCRIPTION
**Pull request by GoelPratyush99**
## Describe your changes
Added new Exec format for kubeconfig file and some classes supporting it.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-ALL: `trigger tiAll`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43976)

### CODE REVIEW

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/kubeconfig/Args.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+package io.harness.k8s.model.kubeconfig;
+
+import io.harness.annotations.dev.HarnessTeam;
+import io.harness.annotations.dev.OwnedBy;
+
+@OwnedBy(HarnessTeam.CDP)
+public interface Args {
+  default String getArgs() {
+    return null;
```
Comment (abosii-harness) [inline 970-api-services-beans/src/main/java/io/harness/k8s/model/kubeconfig/Args.java:16]: Why do we require default here? Should we return then "[]" as default value?

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/kubeconfig/Exec.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@JsonIgnoreProperties(ignoreUnknown = true)
+@OwnedBy(HarnessTeam.CDP)
+public class Exec {
+  @JsonProperty(K8sConstants.KUBECFG_API_VERSION) private String apiVersion;
+  @JsonProperty(K8sConstants.KUBECFG_ARGS) private Args args;
```
Comment (abosii-harness) [inline 970-api-services-beans/src/main/java/io/harness/k8s/model/kubeconfig/Exec.java:25]: This should be List<String>

**Review by abosii-harness — CHANGES_REQUESTED** [review]

**Review by abosii-harness — APPROVED** [review]
> Reviewed 4 of 7 files at r1, 2 of 3 files at r2, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43976)* status: Some of the checks are too old. 9 file(s) to review. 1 of 2 approvals obtained.

**Review by abhinavsingh1196 — APPROVED** [review]
> Reviewed 4 of 7 files at r1, 2 of 3 files at r2, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43976)* status: 1 of 2 approvals obtained.

**Review by abhinavsingh1196 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43976)* status: 1 of 2 approvals obtained.

**Review by abosii-harness — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43976)* status: :shipit: complete! ✓

**Pull request merged**